File: Problems/Algorithms/1704. Determine If String Halves Are Alike/alike_halves.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

        n = len(s)
        count_left, count_right = 0, 0
        for i in range(n//2):
            count_left = count_left + 1 if s[i] in vowels else count_left
            count_right = count_right + 1 if s[n-i-1] in vowels else count_right

        return count_left == count_right

File: Problems/Algorithms/1704. Determine If String Halves Are Alike/alike-halves.cpp
#include <string>
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool halvesAreAlike(string s) {
        auto countVowels = [](const string& str) {
            unordered_set<char> vowels{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
            int count = 0;
            for (char c : str) {
                if (vowels.count(c) > 0) {
                    count++;
                }
            }
            return count;
        };
        
        int n = s.length();
        int mid = n / 2;

        string first = s.substr(0, mid);
        string second = s.substr(mid);

        return countVowels(first) == countVowels(second);
    }
};
